components/suggest/src/schema.rs

```diff
@@ -731,4 +751,51 @@ PRAGMA user_version=16;
         db_file.run_all_upgrades();
         db_file.assert_schema_matches_new_database();
     }
+
+    /// Test that `clear_database()` works correctly during migrations.
+    ///
+    /// TODO: This only checks `ingested_records` and `rs_cache` for now since
+    /// they're very important, but ideally this would test all tables.
+    #[test]
+    fn test_clear_database() -> anyhow::Result<()> {
+        // Start with the v16 schema.
+        let db_file =
+            MigratedDatabaseFile::new(SuggestConnectionInitializer::default(), V16_SCHEMA);
+
+        // Upgrade to v25, the first version with with `ingested_records` and
+        // `rs_cache` tables.
+        db_file.upgrade_to(25);
+
+        // Insert some ingested records and cache data.
+        let conn = db_file.open();
+        conn.execute(
+            "INSERT INTO ingested_records(id, collection, type, last_modified) VALUES(?, ?, ?, ?)",
+            ("record-id", "quicksuggest", "record-type", 1),
+        )?;
+        conn.execute(
+            "INSERT INTO rs_cache(collection, data) VALUES(?, ?)",
+            ("quicksuggest", "some data"),
+        )?;
+        let _ = conn.close();
```

Comment:
I can't figure out how to declare the return value of this test function in a way that lets me write `conn.close()?` here. The error type (or something like that) of this result is different from the error types of the other results where I am able to use `?`, e.g. `conn.execute()?` right above this.

Reply:
I think you can either do

```rs
    conn.close().map_err(|(_, e)| anyhow::anyhow!("{e}"))?;
```
or simply
```rs
    conn.close().expect("Connection should be closed");
```